core/src/test/java/io/jenkins/pluginhealth/scoring/probes/RenovateProbeTest.java

```diff
@@ -0,0 +1,140 @@
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class RenovateProbeTest extends AbstractProbeTest<RenovateProbe> {
+    @Override
+    RenovateProbe getSpy() {
+        return spy(RenovateProbe.class);
+    }
+
+    @Test
+    void shouldNotRequireRelease() {
+        assertThat(getSpy().requiresRelease()).isFalse();
+    }
+
+    @Test
+    void shouldBeRelatedToCode() {
+        assertThat(getSpy().isSourceCodeRelated()).isTrue();
+    }
+
+    static Stream<Arguments> probeResults() {
+        return Stream.of(
+            arguments(// Nothing
+                Map.of()
+            ),
+            arguments(
+                Map.of(
+                    SCMLinkValidationProbe.KEY, ProbeResult.failure(SCMLinkValidationProbe.KEY, "")
+                )
+            ),
+            arguments(
+                Map.of(
+                    SCMLinkValidationProbe.KEY, ProbeResult.failure(SCMLinkValidationProbe.KEY, ""),
+                    LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+                )
+            ),
+            arguments(
+                Map.of(
+                    LastCommitDateProbe.KEY, ProbeResult.failure(LastCommitDateProbe.KEY, "")
+                )
+            ),
+            arguments(
+                Map.of(
+                    LastCommitDateProbe.KEY, ProbeResult.failure(LastCommitDateProbe.KEY, ""),
+                    SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+                )
+            ),
+            arguments(
+                Map.of(
+                    SCMLinkValidationProbe.KEY, ProbeResult.failure(SCMLinkValidationProbe.KEY, ""),
+                    LastCommitDateProbe.KEY, ProbeResult.failure(LastCommitDateProbe.KEY, "")
+                )
+            )
+        );
+    }
+
+    @SuppressWarnings("unchecked")
+    @ParameterizedTest
+    @MethodSource("probeResults")
+    void shouldRequireValidSCMAndLastCommit(Map<String, ProbeResult> details) {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        when(plugin.getDetails()).thenReturn(details);
+
+        final RenovateProbe probe = getSpy();
+        for (int i = 0; i < 6; i++) {
+            assertThat(probe.apply(plugin, ctx))
+                .usingRecursiveComparison()
+                .comparingOnlyFields("id", "message", "status")
+                .isEqualTo(ProbeResult.error(RenovateProbe.KEY, "renovate does not meet the criteria to be executed on null"));
+
+            verify(probe, never()).doApply(plugin, ctx);
+        }
```

Comment:
I supposed you used a loop here in order to go through the different variations of the ProbeResult map entries? 
This is not how this works: JUnit is going through the different variations of the map, as provided by the `probeResults` method, and provide them as a parameter of the test method. 

So, you don't need t loop here.
```suggestion
        assertThat(probe.apply(plugin, ctx))
            .usingRecursiveComparison()
            .comparingOnlyFields("id", "message", "status")
            .isEqualTo(ProbeResult.error(RenovateProbe.KEY, "renovate does not meet the criteria to be executed on null"));

        verify(probe, never()).doApply(plugin, ctx);
```